scroll-tech/zkevm-circuits
# WIP TOWARDS E2E CONTRACT CALL PROVING!

close in favor of #127 

With this PR  e2e contract calls can be proved.    `set_value & retrieve` in `integration-tests/contracts/greeter/Greeter.sol` used here.


How to test:

0. install docker locally
1. cd integration-tests
4. ./run.sh --steps "setup"
5. ./run.sh --steps "gendata"
6. RUST_LOG=circuits,bus_mapping=trace cargo test --profile release --test circuits --features circuits test_evm_circuit_block_greeter_calls -- --nocapture      # 20 rand getter/setter calls tested 

then you can see

```
test test_evm_circuit_block_greeter_calls ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 6 filtered out; finished in 15.48s
```
